crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -506,16 +503,13 @@ async fn c8y_mapper_child_alarm_with_custom_fragment_mapping_to_c8y_json() {
         .await
         .unwrap();
 
-    let expected_msg = json!({"severity":"MAJOR","type":"temperature_alarm","time":"2023-01-25T18:41:14.776170774Z","text":"Temperature high","externalSource":{"externalId":"external_sensor","type":"c8y_Serial"},"customFragment":{"nested":{"value":"extra info"}}});
-
-    while let Ok(Some(msg)) = messages.next().with_timeout(TEST_TIMEOUT_MS).await {
-        assert_json_include!(actual:serde_json::from_str::<serde_json::Value>(&msg).unwrap(), expected:expected_msg);
-    }
-
+    let expected_msg = r#"{"externalSource":{"externalId":"external_sensor","type":"c8y_Serial"},"severity":"MAJOR","type":"temperature_alarm","time":"2023-01-25T18:41:14.776170774Z","text":"Temperature high","customFragment":{"nested":{"value":"extra info"}}}"#;
+    // Expect converted temperature alarm message
+    mqtt_tests::assert_received_all_expected(&mut messages, TEST_TIMEOUT_MS, &[expected_msg]).await;
     //Clear the previously published alarm
     broker
         .publish_with_opts(
-            "tedge/alarms/major/temperature_alarm",
+            "tedge/alarms/major/temperature_alarm/external_sensor",
```

Comment:
Because there was a typo in the topic, the alarm that was sent on `"tedge/alarms/major/temperature_alarm/external_sensor` had to be cleared sending an empty message on the same topic.